templates/etc/profile.d/tlog.sh.epp

```diff
@@ -0,0 +1,62 @@
+<%- |
+  Stdlib::Absolutepath $users_file = '/etc/security/tlog.users',
+  Stdlib::Absolutepath $app_path   = '/usr/bin/tlog-rec-session'
+| -%>
+# Place this script in /etc/profile.d to automatically hook any login or
+# interactive shell into tlog for a user or group listed in
+# /etc/security/tlog.users
+#
+# Entries in tlog.users should be listed one per line where users are bare
+# words such as `root` and groups are prefixed with a percent sign such as
+# `%root`.
+#
+# Copyright 2018 Trevor Vaughan <tvaughan@onyxpoint.com> - Onyx Point, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+TLOG_USERS="<%= $users_file %>"
+TLOG_CMD="<%= $app_path %>"
+
+tlog_parent(){
+  retval=1
+
+  ppid=`ps --no-headers -o ppid $1`
+
+  if [ $ppid -gt 1 ]; then
+    if `ps --no-headers -o ppid,args $1 | grep -q 'tlog-rec-session'`; then
+      return 0
+    else
+      tlog_parent $ppid
+      retval=$?
+    fi
+
+  fi
+
+  return $retval
+}
+
+if [ -f "${TLOG_USERS}" ]; then
+  if ! `tlog_parent $PPID`; then
+    if `grep -qE "^(%${GROUP}|${USER})$" "${TLOG_USERS}"`; then
```

Comment:
GROUP is not defined anywhere.   I put GROUP="`/usr/bin/id -gn`" in /etc/profile and added group %admin2 (the group for my admin user).  When I logged in it hung up.  it never gave me a prompt.  I hit ctl-c and it dropped me to a command prompt and was logging.

Reply:
I get the ATTENTION! Your session is being recorded! but never drop to the prompt unless I hit ctl-c

Reply:
Sorry, it is actually kickiing me out when I hit ctl-c.  Also when I unfixed the error above (the [[ -gt ]])  I got the message -gt: unary operator expected scrolling across my machine like it was logging in or running the profile over and over and over.

Reply:
Sorry, sometimes it logs we out sometimes it drops to the shell.  I think it has to do with how log I wait until I cancel it.  If I wait longer it will drop me out to a shell.